### ISSUE TITLE
recursion error on windows when running pylint on a path with different capitalization

### ISSUE THREAD

**Issue by DetachHead** [pylint-dev/pylint#10137]
### Bug description

```python
# ./tests/sandpit/__init__.py

from tests.sandpit import foo


def asdf():
    return foo.abc()


def fdsa(bar):
    lambda item: item(bar.title())


# ./tests/sandpit/bar.py

def abc(): ...
```

### Configuration

```ini

```

### Command used

```shell
pylint tests\sandpit\__Init__.py
```

### Pylint output

```python
> pylint tests\sandpit\__Init__.py
C:\Users\user\project\.venv\Lib\site-packages\astroid\transforms.py:118: UserWarning: Astroid was unable to transform Expr(value=<Lambda.<lambda> l.9 at 0x1f740428730>).
Some functionality will be missing unless the system recursion limit is lifted.
From pylint, try: --init-hook='import sys; sys.setrecursionlimit(2000)' or higher.
  warnings.warn(
```

### Expected behavior

no error

### Pylint version

```shell
pylint 3.3.2
astroid 3.3.5
Python 3.13.0 (main, Oct 16 2024, 00:33:24) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

windows 11

### Additional dependencies

```python

```
